server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -339,7 +396,9 @@ public void onFailure(Exception e) {
             } else {
                 responseActionListener = listener;
             }
-            if (sendFullVersion || previousState.nodes().nodeExists(destination) == false) {
+            if (sendRemoteState && destination.isRemoteStateNode()) {
+                sendRemoteClusterState(destination, publishRequest.getAcceptedState(), responseActionListener);
```

Comment:
We will need to check if remote publication is enabled here in addition to remote cluster state enabled